### ISSUE TITLE
Consider limit codebase max line length to 80/100

### ISSUE THREAD

**Issue by jneira** [haskell/haskell-language-server#2091]
* Limit the line length is a usual convention that helps to read the codebase (see serokell haskell style guide f.e: https://github.com/serokell/style/blob/master/haskell.md#line-length (but many other style guides recommend it):

> You should keep maximum line length below 80 characters. If necessary, you may use up to 100 characters, although this is discouraged. You should wrap imports at 100 characters.

It would suppose reach a consensus and start to progresively apply it in dedicated pr's, trying to dont disturb existing work in progress.

//cc @Ailrun, as you have worked fantistaclly about the formatting of the codebase

**Comment by IvanMalison** [comment]
Not a regular contributor (yet) but my preference would be towards the lower side of this (80 chars), but seeing as there are quite a few places in the code base that exceed this quite a lot, I would also be happy to see any limit 100 chars or less.

**Comment by michaelpj** [comment]
Allegedly, it already is: https://github.com/haskell/haskell-language-server/blob/master/.editorconfig#L14

**Comment by jneira** [comment]
Uh, thanks, i forgot about that, unfortunately you have to install an extension in vscode to enable it (and i think other editor does not honour it by default neither)

**Comment by Ailrun** [comment]
I believe the current `stylish-haskell` already uses 80 column limit, so AFAIK our codebase is already formatted with 80 column limit in the cases `stylish-haskell` can handle. Do you know some specific example of out-of-80-column case?

**Comment by IvanMalison** [comment]
@Ailrun I dived in to the code base for the first time yesterday and started reading the "AtPoint.hs" file of ghcide. There were too many lines over 100 chars to count. Here are some examples:

128 chars: https://github.com/haskell/haskell-language-server/blob/903fe64ccdf28dc765335d5b730d87a07b951bf1/ghcide/src/Development/IDE/Spans/AtPoint.hs#L122

125 chars: https://github.com/haskell/haskell-language-server/blob/903fe64ccdf28dc765335d5b730d87a07b951bf1/ghcide/src/Development/IDE/Spans/AtPoint.hs#L127

120 chars: https://github.com/haskell/haskell-language-server/blob/903fe64ccdf28dc765335d5b730d87a07b951bf1/ghcide/src/Development/IDE/Spans/AtPoint.hs#L222

124 chars: https://github.com/haskell/haskell-language-server/blob/903fe64ccdf28dc765335d5b730d87a07b951bf1/ghcide/src/Development/IDE/Spans/AtPoint.hs#L318

**Comment by IvanMalison** [comment]
Does not seem limited to that file though. Here are some more examples over 100 chars:

108: https://github.com/haskell/haskell-language-server/blob/903fe64ccdf28dc765335d5b730d87a07b951bf1/plugins/hls-call-hierarchy-plugin/test/Main.hs#L414

104: https://github.com/haskell/haskell-language-server/blob/903fe64ccdf28dc765335d5b730d87a07b951bf1/ghcide/src/Development/IDE/Core/Debouncer.hs#L42

148: https://github.com/haskell/haskell-language-server/blob/903fe64ccdf28dc765335d5b730d87a07b951bf1/ghcide/src/Development/IDE/Core/Compile.hs#L347

113: https://github.com/haskell/haskell-language-server/blob/903fe64ccdf28dc765335d5b730d87a07b951bf1/ghcide/src/Development/IDE/Core/Compile.hs#L478

**Comment by Ailrun** [comment]
@IvanMalison Thank you for the finding. It looks like stylish-haskell indeed does not support those cases.

**Comment by IvanMalison** [comment]
> @IvanMalison Thank you for the finding. It looks like stylish-haskell indeed does not support those cases.

Huh really? Some of those lines look like they have very simple multi-line versions that I would expect it would support.

**Comment by Ailrun** [comment]
@IvanMalison Yeah, I expected so too, but I just tried those specific examples, and stylish-haskell indeed doesn't change those lines.

**Comment by IvanMalison** [comment]
> @IvanMalison Yeah, I expected so too, but I just tried those specific examples, and stylish-haskell indeed doesn't change those lines.

I'm not super familiar with stylish-haskell. Will it allow you to change them to be multi-line defintions without reverting the changes?

It seems like maybe some diligence on the part of contributors not to make huge lines like these may be needed in addition to stylish haskell.

**Comment by Ailrun** [comment]
@IvanMalison AFAIK, if `stylish-haskell` doesn't support some cases, it won't revert them back too.

**Comment by hasufell** [comment]
I'm against this. Should we make a vote or something? Maybe discuss it on the next HLS meeting.
